### ISSUE TITLE
Regression: map_blocks provides array as tuple

### ISSUE THREAD

**Issue by djhoese** [dask/dask#4285]
This is likely related to #4284 but my situation doesn't seem to be resolved by a smaller input size. In my case `map_blocks` is receiving a tuple like `('add-98ef4914c5346006232dac9722c69498', 0, 0)`. Here is my test case:

```python
from glob import glob
import numpy as np

import xarray as xr
import dask.array as da


def four_element_average_dask(d):
    """Average every 4 elements (2x2) in a 2D array"""
    def _mean4(data, block_id=None):
        print("#############################: ", data, block_id)
        rows, cols = data.shape
        return data

    res = d.data.map_blocks(_mean4, dtype=d.dtype)
    return xr.DataArray(res, attrs=d.attrs, dims=d.dims, coords=d.coords)


size = 100
high_res = xr.DataArray(da.random.random((size, size), chunks=4096))
print(high_res.data.size)
high_res = high_res * 1.0 + 0.0

# Persisting before the calculation fixes it
#high_res = high_res.persist()
high_mean = four_element_average_dask(high_res)
ratio = high_res / high_mean
ratio = ratio.compute()
```

If you comment out the `high_res = high_res * 1.0 + 0.0` line or the `ratio = high_res / high_mean` line (and compute `high_mean` instead), then it computes fine. It's also possible I've now minimized my test case so much that I'm doing something wrong, but I'm not seeing it. Any ideas?

**Comment by djhoese** [comment]
Removing the `block_id` fixes this too. Also I noticed `block_info` is now the suggested method for getting the block id but that also fails.

**Comment by mrocklin** [comment]
> Edit: Also seems to work with #4229.

Right.  I think that the right approach is to wait for that to land.  My apologies for the bug until then.  

Thank you also for raising such a nice issue with a nice reproducible error.

**Comment by djhoese** [comment]
I think this has been fixed since #4229 was merged. I'll reopen/recreate if I run in to this again.
